src/shared/libs/database-client/mongo.database-client.ts

```diff
@@ -0,0 +1,58 @@
+import * as Mongoose from 'mongoose';
+import { inject, injectable } from 'inversify';
+import { setTimeout } from 'node:timers/promises';
+
+import { DatabaseClient } from './database-client.interface.js';
+import { Component } from '../../types/index.js';
+import { Logger } from '../logger/index.js';
+
+const RETRY_COUNT = 5;
+const RETRY_TIMEOUT = 1000;
+
+@injectable()
+export class MongoDatabaseClient implements DatabaseClient {
+  private mongoose: typeof Mongoose;
+  private isConnected: boolean;
+
+  constructor(
+    @inject(Component.Logger) private readonly logger: Logger
+  ) {
+    this.isConnected = false;
+  }
+
+  public isConnectedToDatabase() {
+    return this.isConnected;
+  }
```

Comment:
public get isConnectedToDatabase() {
    return this.mongoose?.instance?.isConnected;
  }